### ISSUE TITLE
[envoy plugin] caller cancelled query execution

### ISSUE THREAD

**Issue by omerlh** [open-policy-agent/opa#1964]
# Expected Behavior
Being able to write unit tests for the envoy plugin policies, with envoy input.

# Actual Behavior
See the example test file [here](https://github.com/omerlh/opa-demo/blob/master/policies/common/tests/service2service_tests.rego). When trying to run the tests, I'm getting the following unclear error:
```
data.system.authz.test_allow_authz_path: ERROR (560.292µs)
  eval_cancel_error: caller cancelled query execution
--------------------------------------------------------------------------------
ERROR: 1/1
``` 

I think it some sort of time out, but I'm not sure. I had it when I first tried to write tests for the policy with the whole Envoy input JSON (which is very big), but it solved once I removed all the fields I don't use.

# Steps to Reproduce the Problem

  1. Run the unit tests.

# Additional Info

**Comment by ashutosh-narkar** [comment]
It would be helpful if you shared the policy itself. [Here]( https://play.openpolicyagent.org/p/dePrzoMvpu) is an example of writing unit tests with Envoy input.

**Comment by omerlh** [comment]
Sorry, the policy is [here](https://github.com/omerlh/opa-demo/blob/master/policies/common/policies/service2service.rego). Thank you!

**Comment by ashutosh-narkar** [comment]
Try: https://play.openpolicyagent.org/p/Genzw3Y5jo

**Comment by omerlh** [comment]
Maybe it's an OPA version issue? I'm running Version: 0.13.0

**Comment by omerlh** [comment]
Maybe it's an OPA version issue? I'm running Version: 0.13.0

**Comment by tsandall** [comment]
@omerlh the test timeout issue could have been due to a bug in the test runner that was fixed in v0.15.0: https://github.com/open-policy-agent/opa/issues/1788. The test timeout (defaults to 5 seconds) except that was the timeout FOR ALL tests to run. The fix makes it apply to each test case (5 seconds should be way more than needed.)

That said, I immediately noticed that your test files take a very long time to load. This is due to a known issue w/ the current parser. https://github.com/open-policy-agent/opa/issues/1251.

I'm going to close this since the issue isn't affecting v0.15.0 and newer.
